analyzers/src/SonarAnalyzer.CSharp/Rules/UnusedPrivateMember.cs

```diff
@@ -100,29 +103,33 @@ private static void NamedSymbolAction(SonarSymbolReportingContext context, HashS
                                           Compilation compilation,
                                           HashSet<ISymbol> privateSymbols,
                                           HashSet<ISymbol> internalSymbols,
-                                          BidirectionalDictionary<ISymbol, SyntaxNode> fieldLikeSymbols)
+                                          BidirectionalDictionary<ISymbol, SyntaxNode> fieldLikeSymbols,
+                                          SonarSymbolReportingContext context)
         {
             if (namedType.ContainingType != null
                 // We skip top level statements since they cannot have fields. Other declared types are analyzed separately.
                 || namedType.IsTopLevelProgram()
                 || namedType.DerivesFromAny(IgnoredTypes)
                 // Collect symbols of private members that could potentially be removed
-                || RetrieveRemovableSymbols(namedType, compilation) is not { } removableSymbolsCollector)
+                || RetrieveRemovableSymbols(namedType, compilation, context) is not { } removableSymbolsCollector)
             {
                 return false;
             }
 
             CopyRetrievedSymbols(removableSymbolsCollector, privateSymbols, internalSymbols, fieldLikeSymbols);
 
             // Collect symbols of private members that could potentially be removed for the nested classes
-            foreach (var declaration in namedType.DeclaringSyntaxReferences.Where(r => !r.SyntaxTree.IsConsideredGenerated(CSharpGeneratedCodeRecognizer.Instance, compilation))
-                                                                           .SelectMany(x => x.GetSyntax().ChildNodes().OfType<BaseTypeDeclarationSyntax>()))
+            foreach (var declaration in namedType.DeclaringSyntaxReferences.Where(r => !r.SyntaxTree.IsConsideredGenerated(
+                CSharpGeneratedCodeRecognizer.Instance,
+                compilation,
+                context.SonarLintXml().AnalyzeRazorCode(context.Compilation.Language)))
+                    .SelectMany(x => x.GetSyntax().ChildNodes().OfType<BaseTypeDeclarationSyntax>()))
```

Comment:
I think this query is to big to keep it inside the `foreach`. Could you move it on top and have a variable for it or even extract it to a method. It will make the code easier to read and understand.